src/provider/states/download_package.rs

```diff
@@ -0,0 +1,137 @@
+use std::path::{Path, PathBuf};
+
+use kubelet::pod::Pod;
+use kubelet::state::prelude::*;
+use kubelet::state::{State, Transition};
+use log::{debug, error, info, warn};
+
+use crate::provider::repository::find_repository;
+use crate::provider::repository::package::Package;
+use crate::provider::states::download_package_backoff::DownloadingBackoff;
+use crate::provider::states::install_package::Installing;
+use crate::provider::PodState;
+
+#[derive(Default, Debug, TransitionTo)]
+#[transition_to(Installing, DownloadingBackoff)]
+pub struct Downloading;
+
+impl Downloading {
+    fn package_downloaded<T: Into<Package>>(
```

Comment:
Checksums not implemented yet, I'd defer this until we have looked at our ci processes and the packages that we release in case this has any bearing on it.